src/AppMenuSearch.cc

```
@@ -257,36 +248,67 @@ void AppMenuSearch::collectSearchCandidates(QMenu *menu, QSet<QMenu *> &visited,
     }
 
-    if (addedNamed) {
-        namedAncestors.removeLast();
+    if (addedAncestor) {
+        ancestors.removeLast();
     }
-    if (addedEnablement) {
-        enablementAncestors.removeLast();
+}
+
+bool AppMenuSearch::matchesAncestorsOrText(const SearchCandidate &candidate, const QStringMatcher &matcher, bool ignoreTopLevel, QHash<QAction *, MatchState> &matchCache) const
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)